core/modules/entity/entity.theme.inc

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * @file
+ * Holds the Entity module's theme functions.
+ */
+
+/**
+ * Process variables for entity.tpl.php.
+ */
+function template_preprocess_entity(&$variables) {
+  $variables['view_mode'] = $variables['elements']['#view_mode'];
+  $variables['entity_type'] = $variables['elements']['#entity_type'];
+  $entity_type = $variables['entity_type'];
+  $variables['entity'] = $variables['elements']['#entity'];
+  $entity = $variables['entity'];
+  $variables['entity_id'] = $entity->id();
+  $info = entity_get_info($entity_type);
+
+  $variables['title'] = check_plain(entity_label($entity_type, $entity));
+
+  $uri = entity_uri($entity_type, $entity);
+  $variables['url'] = $uri && !empty($uri['path']) ? url($uri['path'], $uri['options']) : FALSE;
```

Comment:
The `Entity::uri()` might not always return an `options` key. Before this line: `$uri += array('options' => array());`

Reply:
Fixed